src/WebformSubmissionLogMailer.php

```diff
@@ -0,0 +1,130 @@
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class WebformSubmissionLogMailer {
+  use LoggerChannelTrait;
+
+  /**
+   * The config factory service interface.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface;
+   */
+  protected ConfigFactoryInterface $configFactory;
+
+  /**
+   * The mail manager service.
+   *
+   * @var \Drupal\Core\Mail\MailManager;
+   */
+  protected MailManager $mailManager;
+
+  /**
+   * The language manager service.
+   *
+   * @var \Drupal\Core\Language\LanguageManager;
+   */
+  protected LanguageManager $languageManager;
+
+  /**
+   * The request stack.
+   *
+   * @var \Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected RequestStack $requestStack;
+
+  /**
+   * Mailer constructor.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
+   *   The config factory service interface.
+   * @param \Drupal\Core\Mail\MailManager $mailManager
+   *   The mail manager service.
+   * @param \Drupal\Core\Language\LanguageManager $languageManager
+   *   The language manager service.
+   * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
+   *   The request stack.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, MailManager $mailManager, LanguageManager $languageManager, RequestStack $requestStack) {
+    $this->configFactory = $configFactory;
+    $this->mailManager = $mailManager;
+    $this->languageManager = $languageManager;
+    $this->requestStack = $requestStack;
+  }
+
+  /**
+   * Send "failed job" mail to recipients related to webform.
+   *
+   * @param $webformSubmission
+   *   The webform submission that failed.
+   * @param $context
+   *   The logging context.
+   * @return void
+   */
+  public function sendMails($webformSubmission, $context) {
+    $webform = $webformSubmission->getWebform();
+    $webformSettings = $webform->getThirdPartySetting('os2forms', 'os2forms_webform_submission_log');
+
+    $recipients = $this->getRecipients($webformSettings);
+    $module = 'os2forms_webform_submission_log';
+    $key = 'submission_log_notification';
+    $langCode = $this->languageManager->getCurrentLanguage()->getId();
+
+    $params['title'] = 'Webform submission error - Form name: '. $webform->label() . ', Id: ' . $webformSubmission->serial();
+    $params['from'] = $this->configFactory->get('system.site')->get('mail');
+    $params['message'] = $this->createMessage($webform, $webformSubmission, $context);
+
+    foreach ($recipients as $to) {
+      $result = $this->mailManager->mail($module, $key, $to, $langCode, $params);
+      if ($result['result'] !== true) {
+        $logger = $this->getLogger($module);
+        $logger->error('There was a problem sending your email notification');
+      }
+    }
+  }
+
+  /**
+   * Get recipients.
+   *
+   * @param $webformSettings
+   *
+   * @return false|string[]
+   */
+  private function getRecipients($webformSettings) {
```

Comment:
This function always returns `string[]|array`.

We should steals coding standards setup and checks from https://github.com/itk-dev/os2forms_sync/tree/os2forms_sync#coding-standards and https://github.com/itk-dev/os2forms_sync/blob/os2forms_sync/.github/workflows/pr.yaml, say.

Reply:
Added  github action.